File: React/calendar-app/src/components/Event.js
import './Event.css';
function Event(){
    var event = {
        title:"technical",
        description:"discussion",
        startdatetime : "2023-12-01 09:00:00",
        enddatetime: "2023-12-01 06:00:00",
        notificationdatetime : "2023-12-01 09:00:00",
        location: "Chennai",
        isrecurring:false,
        recurring_frequency:"",
        shareeventwith:"ayeshajasmeen79@gmail.com",
        access:"public",
        categoryId:1,
        email:"dilluchitti@gmail.com"
    }
    return(
        <div className="event">
            <h1>Event</h1>
            <br/>
            Event title : {event.title}
            <br/>
            Event description : {event.description}
            <br/>
            Event startdatetime : {event.startdatetime}
            <br/>
            Event enddatetime : {event.enddatetime}
            <br/>
            Event notificationdatetime : {event.notificationdatetime}
            <br/>
            Event location : {event.location}
            <br/>
            Event IsRecurring : {event.IsRecurring}
            <br/>
            Event recurring_frequency : {event.recurring_frequency}
            <br/>
            Event shareeventwith : {event.shareeventwith}
            <br/>
            Event access : {event.access}
            <br/>
            Event categoryId : {event.categoryId}
            <br/>
            Event email : {event.email}
        </div>
    );
}
export default Event;

File: EventCalendarAppSolution/EventCalendarApp/Models/Event.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace EventCalendarApp.Models
{
    public class Event
    {
        /// <summary>
        /// it is a model class for events
        /// </summary>
        public int Id { get; set; }//identity GUID
        public string Title { get; set; }//title of the event
        public string Description { get; set; }//give information about event
        public string StartDateTime { get; set; }//on which date and on what time the event is occurring
        public string EndDateTime { get; set; }//date and time the event to be ended
        public string NotificationDateTime { get; set; }//on which date and time the notification should be send
        public string? Location { get; set; }//location of an event
        public bool? IsRecurring { get; set; }//is the event is repeating or not
        public string? Recurring_frequency { get; set; }//if event is repeating what is its frequency
        public string? ShareEventWith { get; set; }
        public string? Access { get; set; }//whether the event should be in public or private
        public string Category { get; set; } 
        public string Email { get; set; }//Foreign key of user
        [ForeignKey("Email")]
        public User? User { get; set; }
    }  
}